app/src/firebase/snap/SubmitSnap.ts

```diff
@@ -1,28 +1,6 @@
 import Snap from "./Snap";
 import firebase from "firebase/app";
-import { SetStateAction, Dispatch } from "react";
 
-const SubmitSnap = async (
-    snap: Snap,
-    setError: Dispatch<SetStateAction<String>>,
-    setConfirmation: Dispatch<SetStateAction<Boolean>>
-) => {
-    return new Promise(function (resolve, reject) {
-        const result = firebase
-            .firestore()
-            .collection("snaps")
-            .add(snap)
-            .then(() => {
-                setConfirmation(true);
-                resolve(true);
-            })
-            .catch((e) => {
-                console.log(e);
-                setError(e.toString());
-                setConfirmation(false);
-                reject(false);
-            });
-    });
-};
-
-export default SubmitSnap;
+export default async function submitSnap(snap: Snap) {
+    await firebase.firestore().collection("snaps").add(snap);
```

Comment:
```suggestion
    return await firebase.firestore().collection("snaps").add(snap);
```
It's best practice to return the promise, so that it can be awaited by a consumer if necessary.